### PR TITLE
feat: include CR envs enabled on creation in event and update validation

### PR DESCRIPTION
**Pull request by thomasheartman**
This PR improves the handling of change request enables on project creation in two ways:

1. We now verify that the envs you try to enable CRs for exist before passing them on to the enterprise functionality.
2. We include data about environments and change request environments in the project created events.

### CODE REVIEW

File: src/lib/features/project/project-service.ts [hunk 1]
```diff
@@ -1357,11 +1385,12 @@ export default class ProjectService {
     }
 
     // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
-    removeModeForNonEnterprise(data): any {
+    removePropertiesForNonEnterprise(data): any {
         if (this.isEnterprise) {
             return data;
         }
-        const { mode, ...proData } = data;
+
```
Comment (thomasheartman) [inline src/lib/features/project/project-service.ts:1392]:
```suggestion
```

File: src/lib/features/project/project-service.ts [hunk 2]
```diff
@@ -310,7 +320,25 @@ export default class ProjectService {
...
+                    await enableChangeRequestsForSpecifiedEnvironments(
+                        data.changeRequestEnvironments,
+                    );
+
+                data.changeRequestEnvironments =
+                    changeRequestEnvironments ?? [];
+            } else {
+                data.changeRequestEnvironments = [];
+            }
+        }
```
Comment (thomasheartman) [inline src/lib/features/project/project-service.ts:341]: This is the meat of the change. I've done a few things here:

1. Only call this method if we're on enterprise (and if the flag is enabled)
2. Validate the data in OSS before passing it to the function. That keeps all validation in one place and we can safely assume that all envs passed into the function are valid.
3. Assume that the function will return the list of environments and their required approvals. 
4. If we're on enterprise and you didn't provide any change request environments, then we'll return an empty list to indicate that no envs were set up.

This means that when we call this in enterprise, there's less worrying about logic and that the data is valid. Instead, it's a function that takes an input and yields an output. In addition to putting all the heavy logic in one place, it also makes it easier to test all of this in one spot.

**Review by Tymek — APPROVED** [review]
LGTM. I didn't see a test for trying to add a project with no environments, but I'm not sure if we need it

**Comment by thomasheartman** [comment]
Right! Can you explain what the case is; I'm not sure I understand. No environments or no change request environments?

**Comment by Tymek** [comment]
> Right! Can you explain what the case is; I'm not sure I understand. No environments or no change request environments?

No environments. Not related directly to this PR

**Comment by thomasheartman** [comment]
Okay! Are you asking about missing `environments` or an empty list? Either way, I'm pretty sure they're covered in the e2e tests in enterprise.

**Pull request merged**